### DISCUSSION TITLE
/move request, field "squad" is missing  (in "you" and "snakes" array)

### DISCUSSION THREAD

**Discussion by thomasweitzel** [BattlesnakeOfficial/feedback#85] [Bug Reports]
I've tried to solve my first challenge: Avoid Walls. There's no problem with the CLI, but when I run my battlesnake from my server and register it with the battlesnake server, the requests are missing the field "squad". Everything else is provided correctly.

Since I've mapped the requests to types in Rust, I've to declare the field squad as Option<String>. The documentation doesn't mention that this is an optional field.

My question: Is it correct that the field "squad" can be missing from the request, meaning that it is indeed an optional field?

**Comment by bvanvugt** [comment]
Huh. Great catch! Not sure why this is happening, but we should definitely fix it. Let me see what we can do...

**Comment by bvanvugt** [comment]
Okay, update. We've found the source of this, and agree it's a bug in the server side of the API. We'd like to update this to include the `squad` property in all game modes, similarly to how `hazards` are always available even in game modes that don't spawn them.

That said, we're not going to fix this until Summer League 2021 has wrapped up on July 31st. We don't want to risk API changes (even minor ones) at this time.

For now this has been backlogged for us to update in early August. More updates then!

**Comment by robbles** [comment]
@thomasweitzel The fix for this bug is now live. Thanks for reporting it!
